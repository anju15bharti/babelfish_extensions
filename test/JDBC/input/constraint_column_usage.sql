create database db1;
go

Use db1;
go

create table tbl1(a int, b int, primary key(a));
go

create table tbl2(a int, b int, primary key(a), foreign key(b) references tbl1(a));
go

create schema sc1;
go

create table tbl3 (a int, b int, primary key (a,b));
go

create table sc1.tbl4 (a int, b int, check ( a > 0  and b < 0));
go

create table tbl5 (a int, b int, c int, foreign key(b,c) references tbl3(a,b));
go

SELECT * FROM information_schema.CONSTRAINT_COLUMN_USAGE WHERE TABLE_NAME NOT LIKE 'sys%' ORDER BY CONSTRAINT_NAME;
go

Use master;
go

SELECT * FROM information_schema.CONSTRAINT_COLUMN_USAGE WHERE TABLE_NAME NOT LIKE 'sys%' ORDER BY CONSTRAINT_NAME;
go

create table tbl6 (a int, b int, UNIQUE(b));
go

SELECT * FROM information_schema.CONSTRAINT_COLUMN_USAGE WHERE TABLE_NAME NOT LIKE 'sys%' ORDER BY CONSTRAINT_NAME;
go

drop table tbl6;
go

use db1;
go

drop table tbl2;
go
drop table tbl1;
go
drop table tbl5;
go
drop table tbl3;
go
drop table sc1.tbl4;
go
drop schema sc1;
go

use master
go

drop database db1;
go
