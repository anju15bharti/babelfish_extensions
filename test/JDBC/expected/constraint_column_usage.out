create database db1;
go

Use db1;
go

create table tbl1(a int, b int, primary key(a));
go

create table tbl2(a int, b int, primary key(a), foreign key(b) references tbl1(a));
go

create schema sc1;
go

create table tbl3 (a int, b int, primary key (a,b));
go

create table sc1.tbl4 (a int, b int, check ( a > 0  and b < 0));
go

create table tbl5 (a int, b int, c int, foreign key(b,c) references tbl3(a,b));
go

SELECT * FROM information_schema.CONSTRAINT_COLUMN_USAGE WHERE TABLE_NAME NOT LIKE 'sys%' ORDER BY CONSTRAINT_NAME;
go
~~START~~
nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar
db1#!#dbo#!#tbl1#!#a#!#db1#!#dbo#!#tbl1_pkey
db1#!#dbo#!#tbl2#!#b#!#db1#!#dbo#!#tbl2_b_fkey
db1#!#dbo#!#tbl2#!#a#!#db1#!#dbo#!#tbl2_pkey
db1#!#dbo#!#tbl3#!#a#!#db1#!#dbo#!#tbl3_pkey
db1#!#dbo#!#tbl3#!#b#!#db1#!#dbo#!#tbl3_pkey
db1#!#sc1#!#tbl4#!#a#!#db1#!#sc1#!#tbl4_check
db1#!#sc1#!#tbl4#!#b#!#db1#!#sc1#!#tbl4_check
db1#!#dbo#!#tbl5#!#b#!#db1#!#dbo#!#tbl5_b_c_fkey
db1#!#dbo#!#tbl5#!#c#!#db1#!#dbo#!#tbl5_b_c_fkey
~~END~~


Use master;
go

SELECT * FROM information_schema.CONSTRAINT_COLUMN_USAGE WHERE TABLE_NAME NOT LIKE 'sys%' ORDER BY CONSTRAINT_NAME;
go
~~START~~
nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar
~~END~~


create table tbl6 (a int, b int, UNIQUE(b));
go

SELECT * FROM information_schema.CONSTRAINT_COLUMN_USAGE WHERE TABLE_NAME NOT LIKE 'sys%' ORDER BY CONSTRAINT_NAME;
go
~~START~~
nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar#!#nvarchar
master#!#dbo#!#tbl6#!#b#!#master#!#dbo#!#tbl6_b_key
~~END~~


drop table tbl6;
go

use db1;
go

drop table tbl2;
go
drop table tbl1;
go
drop table tbl5;
go
drop table tbl3;
go
drop table sc1.tbl4;
go
drop schema sc1;
go

use master
go

drop database db1;
go
